aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -1325,23 +1366,23 @@ void publishMessagesFromQueue() {
                             AWSIotMqttMessageDeliveryCallback.MessageDeliveryStatus.Fail,
                             message.getUserData().getUserData());
                 }
-            }
 
-            (new Handler(Looper.getMainLooper())).postDelayed(new Runnable() {
-                @Override
-                public void run() {
-                    if (!mqttMessageQueue.isEmpty()) {
-                        if (connectionState == MqttManagerConnectionState.Connected) {
+                if (mReconnectHandlerThread != null && mReconnectHandler != null) {
```

Comment:
I notice a difference here: The HandleThread you created uses its looper whereas the existing code here uses the looper from the main thread. Do you have any reasoning behind this change?

Reply:
I don't believe that the main thread should be responsible for publishing messages from the Queue since this is network related. Originally the first message was published by the reconnecting thread, while all successive messages where published by the main thread. I changed this so that all queued message will be published by the reconnecting thread.